bears/go/GoVetBear.py

```diff
@@ -25,5 +27,93 @@ class GoVetBear:
+                         check_shadow_variables: bool=False,
+                         check_shifts: bool=False,
+                         check_struct_tag_format: bool=False,
+                         disallow_unreachable_code: bool=False,
+                         disallow_unsafe_pointers: bool=False,
+                         disallow_unused_results: bool=False):
+        """
+        :param disallow_assembly_go_mismatches:
+            Mismatches between assembly files and Go function declarations.
+        :param check_useless_assignments:
+            Check for useless assignments.
+        :param check_sync_package_mistakes:
+            Common mistaken usages of the sync/atomic package.
+        :param check_boolean_operator_mistakes:
+            Mistakes involving boolean operators.
+        :param disallow_bad_build_tags:
+            Badly formed or misplaced +build tags.
+        :param disallow_cgo_pointer_violations:
+            Detect some violations of the cgo pointer passing rules.
+        :param disallow_unkeyed_composites:
+            Composite struct literals that do not use the field-keyed syntax.
+        :param disallow_copying_locks:
+            Locks that are erroneously passed by value.
+        :param check_method_signatures:
+            Non-standard signatures for methods with familiar names, including:
+            Format, GobEncode, GobDecode, MarshalJSON, MarshalXML, Peek,
+            ReadByte,ReadFrom, ReadRune, Scan, Seek, UnmarshalJSON, UnreadByte,
+            UnreadRune, WriteByte, WriteTo
+        :param disallow_nil_function_comparisons:
+            Comparisons between functions and nil.
+        :param check_printf_calls:
+            Suspicious calls to functions in the ``Printf`` family.
+        :param check_range_loop_variables:
+            Incorrect uses of range loop variables in closures.
+        :param check_shadow_variables:
+            Variables that may have been unintentionally shadowed.
+        :param check_shifts:
+            Shifts equal to or longer than the variable's length.
+        :param check_struct_tag_format:
+            Struct tags that do not follow the format understood by
+            ``reflect.StructTag.Get``. Well-known encoding struct tags
+            (json, xml) used with unexported fields.
+        :param disallow_unreachable_code:
+            Unreachable code.
+        :param disallow_unsafe_pointers:
+            Likely incorrect uses of ``unsafe.Pointer`` to convert integers to
+            pointers.
+        :param disallow_unused_results:
+            Calls to well-known functions and methods that return a value that
+            is discarded.
+        """
+        # Every check in Vet is true by default.
+        # Adding a flag set to True enables that check and disables all other
+        # checks.
+        # Adding a flag set to False disables that check and runs all
+        # other checks.
```

Comment:
@jayvdb  I've added a comment here for clarification. 
- Actually, there's also a flag called `-all` in `vet` which is by default set to `true`. Therefore, all checks are performed.
- In this bear, there's no behavior where the defaults i.e. `bool=False` is explicitly setting those flags to `false`.

Reply:
Maybe, the comment is a little misleading. Multiple flags set to true do not hinder with each other. I should update that. :)

Reply:
I meant all **flags** are enabled by default. The ones which are added as a part of this PR.

Reply:
Should I add a comment stating - `multiple checks can be enabled and/or disabled at once`?